src/vs/platform/extensionManagement/common/allowedExtensionsService.ts

```diff
@@ -142,4 +144,28 @@ export class AllowedExtensionsService extends Disposable implements IAllowedExte
 
 		return extensionReason;
 	}
+
+	isTrusted(extension: IGalleryExtension): boolean {
+		const publisher = extension.publisher.toLowerCase();
+		if (this.trustedPublishers.includes(publisher) || this.trustedPublishers.includes(extension.publisherDisplayName.toLowerCase())) {
```

Comment:
Is the display name controlled by the publisher? If so, I wonder if someone could create a publisher to bypass this check by making the display name match an existing publisher's id.

Reply:
Yes. But some publisher names like "Microsoft" (which is being used as trusted name here) is reserved and cannot be taken.

Reply:
Someone could create a publisher with display name `ChrMarti` and the check here would pass if my publisher with id `chrmarti` is already trusted?

Reply:
this.trustedPublishers comes from the product here - https://github.com/microsoft/vscode/pull/238540/files#diff-c3333799952c75c7e9727041b8517903f23bec6e96b645a688128c03976bdfdeR49

We only trust publisher display names that come from the product. So in your example ChrMarti won't be trusted

Reply:
Missed that, do we have checks in place to make sure the names in the product.json are in sync with the reserved display names in the marketplace?

Reply:
Resolved in direct chat.